### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by dimarogkov**
[DEMO LINK](https://dimarogkov.github.io/react_pagination/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -34,62 +55,15 @@ export const App: React.FC = () => {
...
-          >
-            »
-          </a>
-        </li>
-      </ul>
+      <Pagination
+        total={items.length}
+        perPage={perPage}
+        currentPage={currentPage}
+        onPageChange={(page) => setCurrentPage(page)}
```
Comment (etojeDenys) [inline src/App.tsx:62]:
```suggestion
        onPageChange={setCurrentPage}
```
we can simplify it

**Review by etojeDenys — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 2]
```diff
@@ -1,18 +1,37 @@
...
+  };
+
+  const getCountOfItems = (arr: string[]) => {
+    return arr.slice(startItemNumber, endItemNumber);
+  };
+
+  const countOfItems = getCountOfItems(items);
+  const lastCountOfItems = countOfItems[countOfItems.length - 1]
+    .split(' ')
+    .pop();
```
Comment (etojeDenys) [inline src/App.tsx:27]: we can simplify this logic
```
const maxItem = Math.max(items.length, currentPage * perPage)

```

File: src/App.tsx [hunk 3]
```diff
@@ -21,6 +40,8 @@ export const App: React.FC = () => {
             data-cy="perPageSelector"
             id="perPageSelector"
             className="form-control"
+            defaultValue={perPage}
+            onChange={({ target }) => changePerPage(target)}
```
Comment (etojeDenys) [inline src/App.tsx:44]: let's pass whole event or value of this target
```suggestion
            onChange={({ target }) => changePerPage(target.value)}
```

File: src/components/Pagination/Pagination.tsx [hunk 4]
```diff
@@ -1 +1,79 @@
...
+
+  const getPageNumbersArr = (count: number) => {
+    const arr = Array.from(Array(count + 1).keys());
+
+    arr.shift();
+
+    return arr;
+  };
+
+  const pageNumbersArr = getPageNumbersArr(totalPages);
```
Comment (etojeDenys) [inline src/components/Pagination/Pagination.tsx:29]: we can simplify it
```
const pages = getNumbers(1, Math.ceil(total / perPage));
```

**Review by etojeDenys — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 2 above)
Comment (dimarogkov) [reply src/App.tsx:27]: Цей код не працює

File: src/components/Pagination/Pagination.tsx (hunk 4 above)
Comment (dimarogkov) [reply src/components/Pagination/Pagination.tsx:29]: До чого цей код?

**Review by mykhalenych — CHANGES_REQUESTED** [review]
Screenshot 2023-09-06 at 18 25 55

fix tests, if you have questions add them to chat

**Pull request closed without merging**